Tools/Scripts/libraries/webkitcorepy/webkitcorepy/task_pool.py

```diff
@@ -139,6 +139,9 @@ def __init__(self, outgoing=None, incoming=None):
         self.outgoing = outgoing or multiprocessing.Queue()
         self.incoming = incoming or multiprocessing.Queue()
 
+        self.outgoing.cancel_join_thread()
+        self.incoming.cancel_join_thread()
+
```

Comment:
I'm assuming there is a specific situation that motivates this change, would you mind sharing the details?

The one thing that gives me pause about this approach is that we risk losing data, and we pass logging information between processes in these queues, which would make data loss difficult to reason about. Data loss is probably better than deadlock, assuming the data loss is rare.

(For others, https://docs.python.org/3/library/multiprocessing.html is Python's documentation on this, no doubt @obyknovenius is looking at it, as am I)

Reply:
The problem this PR is trying to solve is that exiting earlier from `run-webkit-tests` when a lot of tests are still in the queue causes a hang. See the bug report: https://bugs.webkit.org/show_bug.cgi?id=247843. It's easy to reproduce with `run-webkit-tests --exit-after-n-failures=2`. When the number of failing tests exceeds the threshold we kill all the workers and there is no one left to clean up the `outgoing` queue. I updated the PR to only cancel the `TaskPool`s `outgoing` queue when `_ChildException` is thrown.